src/Updater.php

```diff
@@ -177,20 +172,72 @@ public function executeUpdate($module, $update_definition_name) {
         $new_config = NestedArray::mergeDeep($new_config, $update_actions['add']);
       }
 
-      if ($this->updateConfig($configName, $new_config, $expected_config, $delete_keys)) {
+      if ($this->updateConfig($configName, $new_config, $configChange['expected_config'], $delete_keys)) {
         $this->logger->info($this->t('Configuration @configName has been successfully updated.', ['@configName' => $configName]));
       }
       else {
-        $successful = FALSE;
-        $this->logger->warning($this->t('Unable to update configuration for @configName.', ['@configName' => $configName]));
+        $this->registerWarning($this->t('Unable to update configuration for @configName.', ['@configName' => $configName]));
       }
     }
+  }
 
-    // Dispatch event after update has finished.
-    $event = new ConfigurationUpdateEvent($module, $update_definition_name, $successful);
-    $this->eventDispatcher->dispatch(UpdateHelperEvents::CONFIGURATION_UPDATE, $event);
+  /**
+   * Register warning with logging.
+   *
+   * @param string $message
+   *   The message used for logging of warning.
+   */
+  protected function registerWarning($message) {
+    $this->warningCount++;
 
-    return $successful;
+    $this->logger->warning($message);
```

Comment:
Let's add a comment to the $logger property telling people to use this method and not the logger directly.

Reply:
Done - and I have added `logInfo` - so we have functions for both logging levels we are using.